Added options to enable media player for activity groups

albaintor opened the pull request:
Added 2 options (enabled by default) to have :
1/ One global media player entity (as today)
2/ One media player entities per activity group

JackJPowell commented on custom_components/unfoldedcircle/pyUnfoldedCircleRemote/const.py:
Restore this file. 
Eases development while building new functionality into the api.
Final release will switch back to pypi package.
(If there is a better way to handle this scenario, please share)

JackJPowell commented on custom_components/unfoldedcircle/pyUnfoldedCircleRemote/remote_websocket.py:
Restore this file. 
Eases development while building new functionality into the api.
Final release will switch back to pypi package.
(If there is a better way to handle this scenario, please share)

albaintor replied:
Ok done

albaintor commented:
Hi, I don't understand your comment :

currently you have one media player for ALL activities, right ? I added an option (enabled by default) to enable or disable this
I added another option to enable one media player for each activity group
So if you enable option 1 : you will have one media player for sure (for all activities) like today
If you enable option 2 : you will have one media player for each activity group
If you enable both : you will have 1 + n media players where n = number of activity groups

JackJPowell commented on custom_components/unfoldedcircle/config_flow.py:
```diff
@@ -307,23 +308,48 @@ async def async_step_reauth_confirm(
+                    vol.Optional(
+                        CONF_ACTIVITY_GROUP_MEDIA_ENTITIES,
+                        default=self.config_entry.options.get(
+                            CONF_ACTIVITY_GROUP_MEDIA_ENTITIES, True
```
This still leaves a gap for those that would like multiple media players, but don't have activity groups defined. (Or do have some defined but also have stand alone activities) If a single media player solved this case, there wouldn't be a need to create them for each activity group.

albaintor replied:
But in that case the first option (enabled by default) will address it : we keep what I called the "global media player entity" which will address all the activities defined (wether they are in an activity group or standalone activities) and then option 2 to get additional media player entities for each activity group. If you don't have any activity group you should get one media player entity at least.

However improvements need to be done on the selection of the "right" media player entity within an activity. The rules that I defined are....hmmm could be better I think

albaintor replied:
By the way I saw but I may have misunderstood that you are using the source list of the media player entity to switch from one media player to another. Am I correct ? This sounds like a good idea however it raises a regression : media players do have source list most of the time and you loose this ability to select the source of the media player. For apple TV like Shield TV it will be the app (Netflix, Prime...). For my STB it is the TV channel.
I don't have any idea to workaround this

JackJPowell commented:
Not quite following, but it sounds like we agree. Just include those files back in a commit so they are available for easier development while relevant.

albaintor commented:
Hi again Jack,
I have just committed another proposition to handle :
1/ The source selection for a media player entity (for media streamers like Apple TV or Android TV it will be the app like netflix..., for a TV it will be HDMI1, 2..., for a STB it will be the TV channel and so on) => this is important to keep it here I think as this is a must have to handle those commands when you don't have the remote next to you
2/ The media entity selection that you put in the source list : first I removed duplicates, also I moved it and enable the sound mode selection as the list of media entity selection
3/ In the media entity list : when user select an entity, you cannot unselect it to move to automatic selection, so I added an automatic selection entry in the drop down to get batck to automatic mode
4/ Not sure about that : I removed the condition you apply on the user media entity selected when it is off or not playing. There some situations where it is not playing but active, or where it is off and you want to put it back on (when something wrong happened)

As an idea, because sound mode is not ideal to select media player entities : building a sub menu in the source list where you would have : 
- Media player entity -> selection causes to create a new source list built from entities list
- <then the real source list of the media player entity> source 1
-  source 2
- ...
